### PR TITLE
CLOUDP-163155: Improve chocolatey process to help with debugability

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-163155

## Further comments
Simplify chocopkg by 
- Improve template loading by using the [embed](https://pkg.go.dev/embed) lib and no longer take an input folder
- replace the `srcPath` path which was used bot for input and output with the `out` option which now defaults to the `dist` folder like other packaging tasks

Improve the relaase tasks by
- moving the choco package publishing out of the generate msi script to avoid it being conditional and to only be called by the release task as oppose for both release and snapshots
- Account for changes on chocopkg now defaulting to the dist folder
- Attach all generated files to the task to help debug (in particular for sh256 issues we are seeing)

### CODE REVIEW

File: tools/chocolateypkg/chocolateypkg.go [hunk 1]
```diff
@@ -58,26 +65,22 @@ func checkError(err error) {
 	}
 }
 
+//go:embed mongodb-atlas.nuspec
+var atlasNuSpec string
```
Comment (gssbzn) [inline tools/chocolateypkg/chocolateypkg.go:69]: save some file operations by using embedded templates

File: tools/sign/main.go [hunk 2]
```diff
@@ -119,7 +119,7 @@ func sign(notaryURL, filePath, notarySigningKey, notarySigningComment, notaryAut
 	}
 
 	downloadURL := jsonResponse["permalink"].(string)
-	os.TempDir()
```
Comment (gssbzn) [inline tools/sign/main.go:122]: drive-by, seems like leftover

File: build/ci/release.yml [hunk 3]
```diff
@@ -577,16 +610,47 @@ tasks:
       - func: "generate msi"
         vars:
           TOOL_NAME: ${tool_name}
-          RELEASING: true
+      - func: "update choco"
+        vars:
+          TOOL_NAME: ${tool_name}
```
Comment (gssbzn) [inline build/ci/release.yml:615]: can't really test this without releasing so need to confirm if republishing old releases is safe

File: build/ci/release.yml [hunk 4]
```diff
@@ -110,14 +110,28 @@ functions:
...
           SECRET_API_KEY: ${chocolatey_api_key}
         command: bash.exe -c build/package/generate-msi.sh
+  "update choco":
+    - command: subprocess.exec
+      type: test
+      params:
+        <<: *go_options
+        add_to_path:
+          - ${go_bin}
+          - ${go_base_path}${workdir}/bin
```
Comment (wtrocki) [inline build/ci/release.yml:127]: Possible new env var

File: build/package/chocoupdate.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -Eeou pipefail
+
+GOCACHE="$(cygpath --mixed "${workdir:?}\.gocache")"
+CGO_ENABLED=0
+export GOCACHE
```
Comment (wtrocki) [inline build/package/chocoupdate.sh:20]: export can be done with assignment?

**Review by wtrocki — APPROVED** [review]

File: build/package/chocoupdate.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply build/package/chocoupdate.sh:1]: not really unless you have a windows machine

File: build/package/chocoupdate.sh (hunk 5 above)
Comment (gssbzn) [reply build/package/chocoupdate.sh:20]: shellcheck doesn't like that 
https://github.com/koalaman/shellcheck/wiki/SC2155
